### ISSUE TITLE
Bug Report: Parsing Problems

### ISSUE THREAD

**Issue by DevelopDaily** [softwareQinc/qpp#81]
Here are some simple test cases:

```
OPENQASM 2.0;
include "qelib1.inc";

qreg q[2];

h q;

// Case 1: Crashes
// U(+ 0.1 - 0.4  , 0, 0) q[1];
// U(+0.1 - 0.4 , 0, 0) q[1];

// Case 2: Wrong final states 
// U(- 0.4 + 0.1 , 0, 0) q[1];
// U(-0.4 + 0.1 , 0, 0) q[1];

// Case 3: Correct
// U(0.1 - 0.4 , 0, 0) q[1];

// Case 4: Correct
U(-0.3 , 0, 0) q[1];
```

Essentially, all the cases should produce the same final state as the **Case 4** does.

Just a friendly reminder: a similar issue was fixed yesterday at `staq`.

**Comment by vsoftco** [comment]
@DevelopDaily Can you please point me to the corresponding issue in `staq`?

**Comment by DevelopDaily** [comment]
I am not sure how he fixed it. It seems `qpp `has already got that line of code in the commit you mentioned, yet still crashes.

**Comment by meamy** [comment]
Case 2 appears to be fixed in the latest version.

For case 1, the ```qasm``` parser doesn't allow '+' in front of integers/floats. The easiest way to support this is to add a unary '+' production rule and just shortcut it since it's the identity operator. I'll submit a PR.

**Comment by vsoftco** [comment]
@DevelopDaily Fixed by #85
